Parse/Internal/Commands/PFRESTQueryCommand.m

```diff
@@ -133,30 +134,11 @@ + (NSDictionary *)findCommandParametersWithOrder:(NSString *)order
                 NSMutableArray *newArray = [NSMutableArray array];
                 for (PFQuery *subquery in array) {
                     // TODO: (nlutsenko) Move this validation into PFQuery/PFQueryState.
-                    if (subquery.state.limit >= 0) {
-                        [NSException raise:NSInvalidArgumentException
-                                    format:@"OR queries do not support sub queries with limits"];
-                    }
-
-                    if (subquery.state.skip > 0) {
-                        [NSException raise:NSInvalidArgumentException
-                                    format:@"OR queries do not support sub queries with skip"];
-                    }
-
-                    if ([subquery.state.sortKeys count]) {
-                        [NSException raise:NSInvalidArgumentException
-                                    format:@"OR queries do not support sub queries with order"];
-                    }
-
-                    if ([subquery.state.includedKeys count] > 0) {
-                        [NSException raise:NSInvalidArgumentException
-                                    format:@"OR queries do not support sub-queries with includes"];
-                    }
-
-                    if (subquery.state.selectedKeys) {
-                        [NSException raise:NSInvalidArgumentException
-                                    format:@"OR queries do not support sub-queries with selectKeys"];
-                    }
+                    PFParameterAssert(subquery.state.limit == -1, @"OR queries do not support sub queries with limits");
+                    PFParameterAssert(subquery.state.skip == 0, @"OR queries do not support sub queries with skip");
+                    PFParameterAssert(subquery.state.sortKeys.count == 0, @"OR queries do not support sub queries with order");
+                    PFParameterAssert(subquery.state.includedKeys.count == 0, @"OR queries do not support sub-queries with includes");
+                    PFParameterAssert(subquery.state.selectedKeys == nil, @"OR queries do not support sub-queries with selectKeys");
```

Comment:
Should be checking for nil, since empty array there would mean that we should not fetch any keys explicitly.